cmd/operator_install.go

```diff
@@ -0,0 +1,266 @@
+
+}
+
+func runOperatorInstallCmd(ctx context.Context, cmd *cobra.Command) error {
+	pathToConfig, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return fmt.Errorf("failed to parse --config flag: %w", err)
+	}
+
+	if pathToConfig != "" {
+		fmt.Printf("the %s config is being used.\n", pathToConfig)
+		configHandler, err := utils.NewConfigHandler(pathToConfig)
+		if err != nil {
+			return fmt.Errorf("failed to initiate config handler: %w", err)
+		}
+
+		err = configHandler.Handle(ctx, installOperator)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("operator installation finished succesfully.\n")
+		return nil
+
+	}
+
+	fmt.Printf("config wasn't provided. The default config is being used.\n")
+
+	defaultConfigBytes, err := json.Marshal(operator.DefaultInstallationConfig())
+	if err != nil {
+		return fmt.Errorf("failed to marshal default operator manager config: %w", err)
+	}
+
+	defaultConfigData := map[string]interface{}{}
+	if err := json.Unmarshal(defaultConfigBytes, &defaultConfigData); err != nil {
+		return fmt.Errorf("failed to unmarshal default operator manager config: %w", err)
+	}
+
+	err = installOperator(ctx, defaultConfigData)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("operator installation finished succesfully.\n")
+	return nil
+}
+
+func validateOperatorInstallFlags(cmd *cobra.Command) error {
+	pathToConfig, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return fmt.Errorf("failed to parse --config flag: %w", err)
+	}
+
+	if cmd.Flags().Lookup("config").Changed == false {
+		return nil
+	}
+
+	if pathToConfig == "" {
+		return fmt.Errorf("path to config file must be specified")
+	}
+
+	fileExtensionWithDot := filepath.Ext(pathToConfig)
+	if err := utils.IsFileTypeSupported(fileExtensionWithDot); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installOperator(ctx context.Context, data map[string]interface{}) error {
+
+	config, err := operator.NewInstallationConfig(data)
+	if err != nil {
+		return fmt.Errorf("failed to initialize installation config: %w", err)
+	}
+
+	if err := fetchAndApplyManifests(ctx, config); err != nil {
+		return fmt.Errorf("failed to fetch and apply installation manifests: %w", err)
+	}
+
+	operatorManagerConfigMap, err := buildOperatorManagerConfigMap(config)
+	if err != nil {
+		return fmt.Errorf("failed to build operator manager ConfigMap: %w", err)
+	}
+
+	resource, err := convertOperatorManagerConfigMap(operatorManagerConfigMap)
+	if err != nil {
+		return fmt.Errorf("failed to convert operator manager ConfigMap: %w", err)
+	}
+
+	applyHandler := cluster.BaseApplyHandler{}
+	if err := applyHandler.Apply(resource); err != nil {
+		return fmt.Errorf("failed to apply operator manager ConfigMap: %w", err)
+	}
+
+	return nil
+}
+
+func fetchAndApplyManifests(ctx context.Context, config *operator.InstallationConfig) error {
+	manifestSets, err := oceancd.InstallOperator(ctx, operator.NewInstallationPayload(config))
```

Comment:
extract `operator.NewInstallationPayload(config)` to a variable
```
payload := operator.NewInstallationPayload(config)
manifestSets, err := oceancd.InstallOperator(ctx, payload)
```

Reply:
why do we need this?

Reply:
part of Spot standards